fetcher/pivnet.go

```diff
@@ -27,58 +28,68 @@ type Pivnet struct {
 	Client *http.Client
 }
 
+func CreateNewPivnetService() *Pivnet {
+	return &Pivnet{}
+}
+
 func (pivnet *Pivnet) SetToken(token string) {
 	pivnet.UAAAPIToken = token
 }
 
-func (pivnet *Pivnet) Versions(slug string) ([]string, error) {
+func (pivnet *Pivnet) StemcellVersion(slug string, majorStemcellVersion string) (string, error) {
 	if slug == "" {
-		return nil, ErrProductSlugMustNotBeEmpty
+		return "", ErrProductSlugMustNotBeEmpty
+	}
+
+	if majorStemcellVersion == "" {
+		return "", ErrStemcellMajorVersionMustNotBeEmpty
 	}
+
 	locator := url.URL{
 		Scheme: "https",
 		Host:   "network.pivotal.io",
-		Path:   path.Join("/api/v2/products", string(slug), "releases"),
+		Path:   fmt.Sprintf("/api/v2/products/%s/releases/latest?major=%s", slug, majorStemcellVersion),
+	}
```

Comment:
💯   Nice! I like using `url.URL`. It is uncommon but I feel it is easier to follow than a really long `fmt.Sprintf`.

The RawQuery is a bit oddly named but it would be where you put the query for this URL instead of appending to Path.

Reply:
That is cool to know. I added the RawQuery.  I ran into issue with the query parameter adding additional / to the value. So have to use fmt.Sprintf.